src/envoy/utils/authn.cc

```diff
@@ -34,6 +34,12 @@ static void setKeyValue(::google::protobuf::Struct& data, std::string key,
   (*data.mutable_fields())[key].set_string_value(value);
 }
 
+// Helper function to set a key list pair into Struct.
+static void setListValue(::google::protobuf::Struct& data, std::string key,
+                        ::google::protobuf::ListValue* value) {
+  (*data.mutable_fields())[key].set_allocated_list_value(value);
```

Comment:
are you sure of using set_allocated_xxx?  My understanding is it will own the "value" pointer,

Reply:
You are reviewing an older version of the code. The current code does not have set_allocated_xxx.